### ISSUE TITLE
Expose option for configuring user-debug or eng builds

### ISSUE THREAD

**Issue by richnetdesign** [hashbang/aosp-build#24]
For developers or recovery purposes, would be helpful to allow configuring options in aosp-build to make eng or user-debug builds.

This is how it can be done manually
`TARGET_BUILD_TYPE=debug breakfast sargo eng `
`TARGET_BUILD_TYPE=debug brunch sargo eng`

Discussed here
https://source.android.com/setup/build/building

**Comment by ypid** [comment]
> would be helpful to allow configuring options in aosp-build to make eng or user-debug builds

Yes, that would be helpful. I guess for any serious person building and maintaining a custom ROM, a second phone running user-debug or so is basic. I currently only partly have a second phone that is the same as my main one laying around, but I definitional plan to get one again for the Android 11 update.

So, feel free to implement this.

**Comment by ypid** [comment]
Currently, this can only be configured using:

https://github.com/hashbang/aosp-build/blob/a5626601a12751d276e2e131ca3082452bd5fa6a/config/config.yml#L4

I am not sure what the best way to expose this is.

Pro `config.yml`:

* `variant` will change the build output so it should be contained in the config.

Pro for `make` parameter:

* Simpler to make a user debug build.

For the time being, I would propose to leave it as it is unless more compelling arguments for exposing it as `make` parameter are found.
